<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>customquery</class>
 <widget class="QMainWindow" name="customquery">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>600</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>BiblioteQ: Custom Query</string>
  </property>
  <property name="windowIcon">
   <iconset resource="../Icons/icons.qrc">
    <normaloff>:/book.png</normaloff>:/book.png</iconset>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout">
    <item>
     <widget class="QFrame" name="frame">
      <layout class="QVBoxLayout">
       <item>
        <widget class="QLabel" name="label">
         <property name="text">
          <string>This panel is intended for advanced users who have an intimate knowledge of SQL and the database layout of BiblioteQ. Please note that retrieving image information (front_cover, back_cover) will burden the query. Please include myoid and type fields as these fields are required by BiblioteQ for editing of items.</string>
         </property>
         <property name="wordWrap">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_2">
         <property name="text">
          <string>DELETE and SELECT queries are allowed.</string>
         </property>
         <property name="wordWrap">
          <bool>true</bool>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QSplitter" name="splitter">
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
      <property name="childrenCollapsible">
       <bool>false</bool>
      </property>
      <widget class="QGroupBox" name="tables_gb">
       <property name="title">
        <string>Database Tables</string>
       </property>
       <layout class="QVBoxLayout">
        <item>
         <widget class="QTreeWidget" name="tables_t">
          <property name="tabKeyNavigation">
           <bool>true</bool>
          </property>
          <property name="alternatingRowColors">
           <bool>true</bool>
          </property>
          <property name="selectionMode">
           <enum>QAbstractItemView::NoSelection</enum>
          </property>
          <property name="verticalScrollMode">
           <enum>QAbstractItemView::ScrollPerPixel</enum>
          </property>
          <property name="uniformRowHeights">
           <bool>true</bool>
          </property>
          <property name="sortingEnabled">
           <bool>true</bool>
          </property>
          <property name="animated">
           <bool>true</bool>
          </property>
          <property name="columnCount">
           <number>0</number>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QGroupBox" name="query_gb">
       <property name="title">
        <string>Query Statement</string>
       </property>
       <layout class="QVBoxLayout">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout">
          <item>
           <widget class="QLabel" name="label_3">
            <property name="text">
             <string>Saved Queries</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="saved_queries">
            <property name="sizeAdjustPolicy">
             <enum>QComboBox::AdjustToContents</enum>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QTextEdit" name="query_te">
          <property name="tabChangesFocus">
           <bool>true</bool>
          </property>
          <property name="placeholderText">
           <string>Please provide a query statement.</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
    <item>
     <layout class="QHBoxLayout">
      <item>
       <widget class="QPushButton" name="execute_pb">
        <property name="minimumSize">
         <size>
          <width>125</width>
          <height>0</height>
         </size>
        </property>
        <property name="text">
         <string>&amp;Execute</string>
        </property>
        <property name="icon">
         <iconset resource="../Icons/icons.qrc">
          <normaloff>:/16x16/ok.png</normaloff>:/16x16/ok.png</iconset>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="refresh_pb">
        <property name="minimumSize">
         <size>
          <width>125</width>
          <height>0</height>
         </size>
        </property>
        <property name="text">
         <string>&amp;Refresh</string>
        </property>
        <property name="icon">
         <iconset resource="../Icons/icons.qrc">
          <normaloff>:/16x16/reload.png</normaloff>:/16x16/reload.png</iconset>
        </property>
        <property name="shortcut">
         <string>F5</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="save">
        <property name="minimumSize">
         <size>
          <width>125</width>
          <height>0</height>
         </size>
        </property>
        <property name="toolTip">
         <string>Save the query statement.</string>
        </property>
        <property name="text">
         <string>&amp;Save</string>
        </property>
        <property name="icon">
         <iconset resource="../Icons/icons.qrc">
          <normaloff>:/16x16/filesave.png</normaloff>:/16x16/filesave.png</iconset>
        </property>
       </widget>
      </item>
      <item>
       <spacer>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QPushButton" name="close_pb">
        <property name="minimumSize">
         <size>
          <width>125</width>
          <height>0</height>
         </size>
        </property>
        <property name="text">
         <string>&amp;Close</string>
        </property>
        <property name="icon">
         <iconset resource="../Icons/icons.qrc">
          <normaloff>:/16x16/cancel.png</normaloff>:/16x16/cancel.png</iconset>
        </property>
        <property name="shortcut">
         <string>Ctrl+W</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
 </widget>
 <tabstops>
  <tabstop>tables_t</tabstop>
  <tabstop>saved_queries</tabstop>
  <tabstop>query_te</tabstop>
  <tabstop>execute_pb</tabstop>
  <tabstop>refresh_pb</tabstop>
  <tabstop>save</tabstop>
  <tabstop>close_pb</tabstop>
 </tabstops>
 <resources>
  <include location="../Icons/icons.qrc"/>
 </resources>
 <connections/>
</ui>
